### PR TITLE
Be more prettier

### LINKED ISSUE #11474: Blocks migration: figure out Prettier in Jetpack
Jetpack Gutenberg blocks migration to Jetpack and our build there doesn't depend on Prettier per-se, but would be good to ensure Prettier continues to work for blocks and share same Prettier config between projects as a package.

[Packaging docs](https://github.com/Automattic/wp-calypso/blob/ce2493f7fcb3662d16dda09d8885620f0051728b/docs/monorepo.md#readme)

https://prettier.io/docs/en/configuration.html

`.prettierrc` configs:
- [in Jetpack](https://github.com/Automattic/jetpack/blob/ada24ea2b6e173f77ecb4f9c32393d72554013cc/.prettierrc)
- [in Calypso](https://github.com/Automattic/wp-calypso/blob/ad3e86832a0cec80b1c341f82e2a14e8fef62720/.prettierrc)

Related previous work in Jetpack https://github.com/Automattic/jetpack/pull/11007

See also https://github.com/Automattic/wp-calypso/issues/31204

cc @brbrr  @jsnajdr

### PR DESCRIPTION
**Pull request by sirreal**
- Use size-optimized build of prettier, see https://github.com/Automattic/wp-calypso/pull/29987 for details. Same behavior. Faster, smaller installs (props @jsnajdr 👏)
- Add compiled files to `.eslintignore`. We're not interested in linting or prettifying compiled files. Prettier is set to use this for its ignore.
- Add `*.min.js` to `.eslintignore`. We don't want to linting or prettify minified JS.
- Run prettier on _all the JS_ (with a few exceptions).

 ✨ Think about it.

## Questions

Do we need to drop jshint in favor of Calypso eslint? Calypso eslint has evolved with prettier so that they complement each other. We drop any lints that are purely formatting (prettier automates this) and only show lints that require developer intervention.

Do we need to audit the differences in the configs (current jshint vs. Calypso eslint)? To the best of my knowledge, there is some JavaScript in Jetpack that is served, as-is, without transpilation. That's very different from Calypso and may have different linting requirements.

One of the prettier options that we're using in Calypso is `es5` trailing comma, like this:

```js
// none (align with jshint rule: Extra comma. (it breaks older versions of IE))
[ 1, 2 ]; { a: 'a' }; console.log( 1 );
// es5 (current option, what Calypso uses)
[ 1, 2, 3, ]; { a: 'a', }; console.log( 1 );
// all (for completeness, we don't want this at this time)
[ 1, 2, 3, ]; { a: 'a', }; console.log( 1, );
```

https://github.com/Automattic/jetpack/blob/0dc222f5382588ffe0d7c92c4f2618e452db7edd/.prettierrc#L5

## Changelog

- yarn (install the updated prettier - should have identical functionality)
- Run prettier on all of the source JavaScript:
  ```
  yarn run reformat-files
  ```

Closes #11474

### CODE REVIEW

File: package.json
Comment (jeherve) [inline package.json]: What's the reasoning behind that change?

File: package.json
Comment (jeherve) [inline package.json]: Should we also include js files in the `_inc` directory? We have a few JavaScript files in `_inc/lib` for example.

**Review by jeherve — COMMENTED** [review]
Well this should be fun to sync back to WordPress.com :)

Is this ready for review?

File: package.json
Comment (sirreal) [reply package.json]: Let's do it all!

File: package.json
Comment (sirreal) [reply package.json]: It's an optimized bundle of our prettier flavor that provides the bin without any excess. Details: https://github.com/Automattic/wp-calypso/pull/29987

**Comment by kraftbj** [comment]
We should see about forcing this in our pre-commit hook for whatever files we prettify: https://github.com/Automattic/jetpack/blob/master/bin/pre-commit-hook.js#L51

**Comment by kraftbj** [comment]
Testing failure is due to multiple jshint errors on legacy files.

**Comment by simison** [comment]
>We should see about forcing this in our pre-commit hook for whatever files we prettify

Here's an example from Calypso: https://github.com/Automattic/wp-calypso/blob/4e38750947bc346e6438744b755c0ced585530cb/bin/pre-commit-hook.js#L51-L63

**Comment by simison** [comment]
>Testing failure is due to multiple jshint errors on legacy files.

@brbrr did you start replacing jshint with eslint as mentioned in Slack?

**Comment by simison** [comment]
>We should see about forcing this in our pre-commit hook

There might be some overlap with https://github.com/Automattic/jetpack/pull/11515

**Comment by sirreal** [comment]
> Well this should be fun to sync back to WordPress.com :)

😅 Have faith! 😉 

> Is this ready for review?

This was a quick one just to gauge interest —or resistance!

I'll update the description with some real details, run on more JS (all of `_inc`) and label for review.

**Comment by sirreal** [comment]
Rebased and ran prettier

**Comment by sirreal** [comment]
> One of the prettier options that we're using in Calypso is es5 trailing comma, like this:

I think we can safely drop the jshint rule and use `es5` trailing commas.

According to MDN, trailing commas are supported:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas#Browser_compatibility

2ality says <=ie8 are problematic. I don't think that's in our support matrix 😁 

http://2ality.com/2013/07/trailing-commas.html

**Comment by sirreal** [comment]
I've looked at jshint docs and our config and I think we're sorely in need of an update. That's a topic for another PR, but I'd like draw some conclusion from conversation around this PR in the meantime 🙇

**Comment by sirreal** [comment]
Rebased and reprettierfied ✨

**Review by lezama — DISMISSED** [review]
@sirreal you can't be more prettier

File: modules/custom-css/custom-css/js/codemirror.min.js
Comment (enejb) [inline modules/custom-css/custom-css/js/codemirror.min.js]: @sirreal This file should probably not be prettified. 
I wonder if we should be just using the code mirror file that gets shipped with core instead?

File: modules/shortcodes/js/jquery.cycle.min.js
Comment (enejb) [inline modules/shortcodes/js/jquery.cycle.min.js]: @sirreal This seems to be a jquery plugin http://jquery.malsup.com/cycle/ but it doesn't seems to be maintained. 

Also not it seems like we are prettifying the minified version here.

File: modules/shortcodes/js/recipes-printthis.js
Comment (enejb) [inline modules/shortcodes/js/recipes-printthis.js]: @sirreal  
This seems to be a jquery plugin as well. 
That could also be updated to a newer version see 
https://github.com/jasonday/printThis/blob/master/printThis.js

**Comment by sirreal** [comment]
Rebased

File: modules/shortcodes/js/recipes-printthis.js
Comment (sirreal) [reply modules/shortcodes/js/recipes-printthis.js]: With the source duplicated here, I don't see much reason not to run prettier on this.

Updating the plugin would be material for another PR.

File: modules/shortcodes/js/jquery.cycle.min.js
Comment (sirreal) [reply modules/shortcodes/js/jquery.cycle.min.js]: This is used for the slideshow:

https://github.com/Automattic/jetpack/blob/abf3988da482778e9d90ddb35b9ee2368a9d905d/modules/shortcodes/slideshow.php#L275

Yep, would be good to leave this as-is. I'll roll it back add this to `.eslintignore` 👍

**Comment by enejb** [comment]
Can we also add `modules/custom-css/custom-css/js/codemirror.min.js`  to the .eslintignore rules for now.

**Comment by sirreal** [comment]
I've rebased and added `*.min.js` to the `.eslintignore`.

File: modules/carousel/jetpack-carousel.js
Comment (sirreal) [inline modules/carousel/jetpack-carousel.js]: I was concerned about this change but I've tested and confirmed that we're getting numbers on both sides 👍

**Comment by sirreal** [comment]
Can we proceed in spite of the wpcom test failing?

@brbrr do we have some e2e tests that we can run against this?

File: modules/carousel/jetpack-carousel.js
Comment (enejb) [reply modules/carousel/jetpack-carousel.js]: Thanks for confirming and checking that!

**Comment by enejb** [comment]
@dereksmart Any idea what the best process is to get this merged on the .com side? 
This PR would help us a lot.

**Comment by scruffian** [comment]
@kraftbj what kind of review do you want? Just a sanity check that these changes won't change the functionality?

**Comment by kraftbj** [comment]
@scruffian Yeah and if anything is questionable at all to verify it is non-functional (at least one strict comparison added was already checked above already, etc).

**Comment by scruffian** [comment]
There's a lot to check there! I found a few issues when looking at the Phabricator diff, but I'm not sure if that's just a problem with the way it's displayed...

**Comment by kraftbj** [comment]
@scruffian With the phab diff, I'm less concerned. We'll likely need to scrap it and do some file-by-file work. Fusion couldn't apply the patch cleanly, so the breakage is likely due to some things out of sync already messing things up.

If you could be another set of eyes that the PR is 🚢 , we can get it merged, then fight the sync battle moving forward so the PR doesn't get moldy/stale.

**Comment by sirreal** [comment]
We might break this PR down into several, it won't take much work but will clarify things greatly.

There are lint fixes here that were prompted by the prettier changes. Those are the riskiest changes and can be isolated.

The eslintignote and prettier dependency changes can also be isolated.

Running prettier can be daunting, but it is very safe. The danger comes from when the large prettier diff obscures hides relevant changes with sheer volume. When isolated into a high volume low risk change, review is easier.

Shall I proceed with that plan?

**Comment by roccotripaldi** [comment]
> Shall I proceed with that plan?

Seems like a solid plan!

File: modules/custom-css/custom-css/js/css-editor.js
Comment (scruffian) [inline modules/custom-css/custom-css/js/css-editor.js]: Another example of `==` being replaced with `===`

**Comment by scruffian** [comment]
This looks good to me except the one comment above.

I also think that these files look like libs, so maybe we don't want to change them:

_inc/postmessage.js
_inc/spin.js
modules/custom-css/custom-css/js/core-customizer-css-preview.js
modules/custom-css/custom-css/js/core-customizer-css.core-4.9.js
modules/custom-css/custom-css/js/core-customizer-css.js

Certainly the bottom of _inc/postmessage.js is a copy of json2...

File: modules/custom-css/custom-css/js/css-editor.js
Comment (sirreal) [reply modules/custom-css/custom-css/js/css-editor.js]: This is not prettier, this was a manual change I did to fix the lint.

Broken out into #11576, I'll remove from this PR

File: modules/carousel/jetpack-carousel.js
Comment (sirreal) [inline modules/carousel/jetpack-carousel.js]: https://github.com/Automattic/jetpack/pull/11577

**Comment by sirreal** [comment]
I've extracted several smaller parts of this that should be safe to land. I'll rebase and we should wind up with a very large but very safe change here the will only run prettier on more files.

- [ ] https://github.com/Automattic/jetpack/pull/11579
- [ ] https://github.com/Automattic/jetpack/pull/11578
- [ ] https://github.com/Automattic/jetpack/pull/11577
- [ ] https://github.com/Automattic/jetpack/pull/11576

File: modules/carousel/jetpack-carousel.js
Comment (sirreal) [reply modules/carousel/jetpack-carousel.js]: This was ignored before, but the ignore line comment was dropped down by prettier.

I think this is the riskiest linter-initiated change out of the bunch and I thank after rebasing on the other fixes I'd prefer to put the ignore comment back on this line.

**Comment by sirreal** [comment]
Dropped the extracted changes and rebased.

**Comment by brbrr** [comment]
> @brbrr did you start replacing jshint with eslint as mentioned in Slack?

@simison Sorry for the delayed response. I was AFK last week. No, I haven't started anything yet.

**Comment by brbrr** [comment]
> Can we proceed in spite of the wpcom test failing?
> 
> @brbrr do we have some e2e tests that we can run against this?

Most of our e2e tests are Calypso related, so I'd say no - we don't have e2e tests on which we can rely on here.

**Comment by sirreal** [comment]
Rebased and conflict free! Let's get pretty!

**Comment by kraftbj** [comment]
There's an error from Travis wrt `yarn lint`. Can you give it a look? https://travis-ci.org/Automattic/jetpack/jobs/508925642

With 7.2 code freeze on Monday EOD, I'd like to merge this as early as possible next week _after_ the 7.2 branch is spun off so it has the most time possible to soak in master.

**Comment by sirreal** [comment]
The lint should be fixed, it's a disable-line comment that was shifted when prettier ran.

👍 Sounds like a good plan to land when the 7.2 branch is created.

**Comment by simison** [comment]
Let's update `.editorconfig` here: 

https://github.com/Automattic/jetpack/blob/5c63eae84cd232c8031c28676db0db0e2362d010/.editorconfig#L16-L18

**Comment by simison** [comment]
Thoughts on updating the pre-commit hook to run on everything here?

https://github.com/Automattic/jetpack/blob/5c63eae84cd232c8031c28676db0db0e2362d010/bin/pre-commit-hook.js#L44-L56

**Comment by sirreal** [comment]
Rebased, addressed feedback:
- Now we prettier json on precommit
- Run prettier on _all the js/json/jsx_ except what's in .eslintignore

Ran prettier again, conflict free.

**Comment by sirreal** [comment]
Rebased. With 7.2 branched, this should be ready to land anytime.

**Comment by ockham** [comment]
Rebased with :heart: and care.

**Comment by sirreal** [comment]
Rebased to fix conflicts

**Comment by ockham** [comment]
Need to rebase again, I'll do that. We should land this soon so we're not trapped in rebase limbo forever.

File: .editorconfig
Comment (simison) [inline .editorconfig]: I think we can remove `.jshintrc` as that file is already using tabs.

File: modules/custom-post-types/js/many-items.js
Comment (simison) [inline modules/custom-post-types/js/many-items.js]: Noting that I don't see these options in https://jshint.com/docs/options/

Otherwise, they seemed like something that would've conflicted with Prettier.

Comment (ockham) [reply modules/custom-post-types/js/many-items.js]: Those options[ were removed](https://github.com/jshint/jshint/releases/tag/2.5.0) from `jshint` 2.5.0 four years ago! :laughing: 

JP is on `jshint` 2.9.7 now, probably fine to remove already.

Comment (sirreal) [reply modules/custom-post-types/js/many-items.js]:
```suggestion
```

**Review by simison — DISMISSED** [review]
It's impossible to test _everything_ this file touches but I've poked around, skimmed the changes and all looks good.

Also ran different build and lint scripts everything runs as expected.

It'll have a month to be in `master` before 7.3 ships so it'll get some more testing that way.

File: modules/custom-post-types/js/many-items.js
Comment (simison) [reply modules/custom-post-types/js/many-items.js]: Doesn't need to be in this PR, seems tedious task as there might be valid options mixed in between.

Seems like easier to remove them all at once together with jshint.

File: .editorconfig
Comment (simison) [reply .editorconfig]: did in https://github.com/Automattic/jetpack/pull/11507/commits/f54fd2bf873ac89ea72961297157365285f78b9f

**Review by kraftbj — DISMISSED** [review]

**Review by kraftbj — APPROVED** [review]
Rebased again, need to run prettier on package.json, but avoiding it trying to avoid more rebasing until Travis lands.

**Pull request merged**
